core/quality/morqa_transport.go

```diff
@@ -50,14 +46,49 @@ func mapEventToMetric(event Event) *metrics.Event {
 	switch event.EventName {
 	case startupEventName:
 		return &metrics.Event{
-			IsProvider: false,
-			TargetId:   "0x1",
 			Metric: &metrics.Event_VersionPayload{
 				VersionPayload: &metrics.VersionPayload{
 					Version: event.Application.Version,
 				},
 			},
 		}
+	case sessionEventName:
+		context := event.Context.(sessionEventContext)
+		return &metrics.Event{
+			Signature:  context.Consumer,
+			TargetId:   context.Provider,
+			IsProvider: false,
+			Metric: &metrics.Event_SessionEventPayload{
+				SessionEventPayload: &metrics.SessionEventPayload{
+					Event: context.Event,
+					Session: &metrics.SessionPayload{
+						Id:             context.ID,
+						ServiceType:    context.ServiceType,
+						ProviderContry: context.ProviderCountry,
+						ConsumerContry: context.ConsumerCountry,
+					},
+				},
+			},
+		}
+	case sessionDataName:
+		context := event.Context.(sessionDataContext)
+		return &metrics.Event{
```

Comment:
add a method on sessionDataContext toEvent() and call it here to avoid this tedious block in the switch

Reply:
done